wandb/sdk/wandb_init.py

```diff
@@ -449,12 +449,67 @@ def init(
     id=None,
     settings: Union[Settings, Dict[str, Any], None] = None,
 ) -> RunBase:
-    """Initialize a wandb Run.
-
+    """Initialize W&B
+    If called from within Jupyter, initializes a new run and waits for a call to
+    `wandb.log` to begin pushing metrics.  Otherwise, spawns a new process
+    to communicate with W&B.
     Args:
-        entity: alias for team.
-        team: personal user or team to use for Run.
-        project: project name for the Run.
+        job_type (str, optional): The type of job running, defaults to 'train'
+        dir (str, optional): An absolute path to a directory where metadata will
+            be stored.
+        config (dict, argparse, or tf.FLAGS, optional): The config parameters
+            (typically hyperparameters) to store with the run.
+        project (str, optional): W&B Project.
+        entity (str, optional): W&B Entity.
+        reinit (bool, optional): Allow multiple calls to init in the same process.
+        tags (list, optional): A list of tags to apply to the run.
+        group (str, optional): A unique string shared by all runs in a given group.
+        name (str, optional): A display name for the run which does not have to be
+            unique.
+        notes (str, optional): A multiline string associated with the run.
+        magic (bool, dict, or str, optional): magic configuration as bool, dict,
+            json string, yaml filename.
+        config_exclude_keys (list, optional): string keys to exclude storing in W&B
+            when specifying config.
+        config_include_keys (list, optional): string keys to include storing in W&B
+            when specifying config.
+        anonymous (str, optional): Can be "allow", "must", or "never". Controls
+            whether anonymous logging is allowed.  Defaults to never.
+        mode (str, optional): Can be "online", "offline" or "disabled". Defaults to
+            online.
+        allow_val_change (bool, optional): allow config values to be changed after
+            setting. Defaults to true in jupyter and false otherwise.
+        resume (bool, str, optional): If set to true, automatically resume the previous
+            run if script is run from the same machine.  If set to a string (runid) wandb
+            will resume the specified run_id.
+            See https://docs.wandb.com/library/advanced/resuming for more detail.
```

Comment:
Let's get rid of the reference to setting to "run_id".  We can keep `True`, this is the same as "auto".  For the string case it can be "allow", "must", "never", or "auto" and in the "allow" or "must" case you need to specify "id" as well.

Reply:
does auto only resume the previous run if it crashed?  What's the default behavior?

Reply:
yes, run can not be successfully "finished" when using auto resume

Reply:
ok looks like default is "never"

Reply:
not really.  default is None
which means we might overwrite the run

Reply:
None is the same as "auto"?  I'll update the docs.

Reply:
None is it's own thing.  By default we won't resume the run, we'll just overwrite whatever run exists if a user specifies an "id" in init and doesn't specify "resume".  I think a more sane behavior would be to make the default "allow", but that's out of scope here.  I would just say the default is to not resume a run if it exists.